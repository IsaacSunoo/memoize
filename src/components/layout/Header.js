import React, { Component } from 'react';
import DropdownMenu from '../DropdownMenu';

class Header extends Component {

    state = {
        showDropdown: false
    }

    displayDropdown = () => {
        this.setState({ showDropdown: !this.state.showDropdown})
    }

    render() {
        return (
          <div>
              <div className='header-container'>
                  <div className='title-menu'>
                      <h1 className='header-title'>MasterPrototypes.</h1>
                        <div className='dropdown-container'>
                            <button type='submit' className='menu-button' onClick={this.displayDropdown}>
                                <i className="fas fa-grip-lines"></i>
                            </button>
                        </div>
                        <div>
                            {
                                this.state.showDropdown ? <DropdownMenu /> : ''
                            }

                        </div>
                  </div>
              </div>
          </div>
        )

    }
}

export default Header;