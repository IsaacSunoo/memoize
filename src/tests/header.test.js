import React from 'react';
import Header from '../components/layout/Header';
import { shallow } from 'enzyme';

describe('App', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Header />
        )
    });

    it('should match the snapshot with all data passed in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have a default state', () => {
        expect(wrapper.state()).toEqual({ showDropdown: false });
    })

    it('should change the state of dropdown', () => {
        expect(wrapper.state('showDropdown')).toEqual(false);
        wrapper.instance().displayDropdown();
        expect(wrapper.state('showDropdown')).toEqual(true);
    })

});