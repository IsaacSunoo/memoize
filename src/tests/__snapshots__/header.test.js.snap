// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<div>
  <div
    className="header-container"
  >
    <div
      className="title-menu"
    >
      <h1
        className="header-title"
      >
        MasterPrototypes.
      </h1>
      <div
        className="dropdown-container"
      >
        <button
          className="menu-button"
          onClick={[Function]}
          type="submit"
        >
          <i
            className="fas fa-grip-lines"
          />
        </button>
      </div>
      <div />
    </div>
  </div>
</div>
`;
